### PR TITLE
docs: run docgen and make upgrade guide update

### PR DESCRIPTION
**Pull request by rschalo**
Fixes #N/A 

**Description**
Updates to docs and run docgen

**How was this change tested?**

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/docs/upgrading/v1-migration.md [hunk 1]
```diff
@@ -52,11 +52,11 @@ The upgrade guide will first require upgrading to your latest patch version prio
 
    The Karpenter version you are running must be between minor version `v0.33` and `v0.37`. To be able to roll back from Karpenter v1, you must rollback to on the following patch release versions for your minor version, which will include the conversion webhooks for a smooth rollback:
 
-   * v0.37.2
-   * v0.36.4
-   * v0.35.7
-   * v0.34.8
-   * v0.33.7
+   * v0.37.3
+   * v0.36.6
```
Comment (engedaam) [inline website/content/en/docs/upgrading/v1-migration.md:56]:
```suggestion
   * v0.36.5
```

File: website/content/en/v1.0/upgrading/v1-migration.md [hunk 2]
```diff
@@ -320,7 +320,7 @@ export KARPENTER_IAM_ROLE_ARN="arn:${AWS_PARTITION}:iam::${AWS_ACCOUNT_ID}:role/
 2. Set Karpenter Version
 
 ```bash
-# Note: v0.33.6 and v0.34.7 include the v prefix, omit it for versions v0.35+
+# Note: v0.33.8 and v0.34.9 include the v prefix, omit it for versions v0.35+
```
Comment (engedaam) [inline website/content/en/v1.0/upgrading/v1-migration.md:323]: nit
```suggestion
# Note: v0.33.x and v0.34.x include the v prefix, omit it for versions v0.35+
```

File: website/content/en/docs/upgrading/upgrade-guide.md [hunk 3]
```diff
@@ -88,7 +88,7 @@ Below is the full changelog for v1, copied from the [v1 Migration Upgrade Proced
 * Karpenter now adds a readiness status condition to the EC2NodeClass. Make sure to upgrade your Custom Resource Definitions before proceeding with the upgrade. Failure to do so will result in Karpenter being unable to provision new nodes.
 * Karpenter no longer updates the logger name when creating controller loggers. We now adhere to the controller-runtime standard, where the logger name will be set as `"logger": "controller"` always and the controller name will be stored in the structured value `"controller"`
 * Karpenter updated the NodeClass controller naming in the following way: `nodeclass` -> `nodeclass.status`, `nodeclass.hash`, `nodeclass.termination`
-* Karpenter's NodeClaim status conditions no longer include the `severity` field
+* Starting with `0.37.3` Karpenter's NodeClaim status conditions no longer include the `severity` field
```
Comment (jonathan-innis) [inline website/content/en/docs/upgrading/upgrade-guide.md:91]: Is this true? Isn't this starting from the first minor version?

Comment (jonathan-innis) [reply website/content/en/docs/upgrading/upgrade-guide.md:91]: I'm assuming you might want to have the same statement around the webhook being enabled here

File: website/content/en/v0.36/upgrading/upgrade-guide.md [hunk 4]
```diff
@@ -25,14 +25,6 @@ If you get the error `invalid ownership metadata; label validation error:` while
 
 * As part of the helm chart [karpenter](https://gallery.ecr.aws/karpenter/karpenter) - [source](https://github.com/aws/karpenter/blob/main/charts/karpenter/crds). Helm [does not manage the lifecycle of CRDs using this method](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/), the tool will only install the CRD during the first installation of the Helm chart. Subsequent chart upgrades will not add or remove CRDs, even if the CRDs have changed. When CRDs are changed, we will make a note in the version's upgrade guide.
 
-In general, you can reapply the CRDs in the `crds` directory of the Karpenter Helm chart:
```
Comment (jonathan-innis) [inline website/content/en/v0.36/upgrading/upgrade-guide.md:28]: Why did we drop this bit?

File: website/content/en/v0.37/upgrading/upgrade-guide.md [hunk 5]
```diff
@@ -25,14 +25,6 @@ If you get the error `invalid ownership metadata; label validation error:` while
 
 * As part of the helm chart [karpenter](https://gallery.ecr.aws/karpenter/karpenter) - [source](https://github.com/aws/karpenter/blob/main/charts/karpenter/crds). Helm [does not manage the lifecycle of CRDs using this method](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/), the tool will only install the CRD during the first installation of the Helm chart. Subsequent chart upgrades will not add or remove CRDs, even if the CRDs have changed. When CRDs are changed, we will make a note in the version's upgrade guide.
 
-In general, you can reapply the CRDs in the `crds` directory of the Karpenter Helm chart:
```
Comment (jonathan-innis) [inline website/content/en/v0.37/upgrading/upgrade-guide.md:28]: Why did we drop this?

File: website/content/en/docs/upgrading/upgrade-guide.md (hunk 3 above)
Comment (engedaam) [inline website/content/en/docs/upgrading/upgrade-guide.md:91]: Why is this starting at 0.37.3?

File: website/content/en/docs/upgrading/upgrade-guide.md [hunk 6]
```diff
@@ -129,6 +131,7 @@ The Ubuntu EKS optimized AMI has moved from 20.04 to 22.04 for Kubernetes 1.29+.
...
 * Karpenter now disallows `nodepool.spec.template.spec.resources` to be set. The webhook validation never allowed `nodepool.spec.template.spec.resources`. We are now ensuring that CEL validation also disallows `nodepool.spec.template.spec.resources` to be set. If you were previously setting the resources field on your NodePool, ensure that you remove this field before upgrading to the newest version of Karpenter or else updates to the resource may fail on the new version.
+* Starting with `0.34.9` Karpenter has enabled conversion webhooks by default to improve the v1 migration experience.
```
Comment (engedaam) [inline website/content/en/docs/upgrading/upgrade-guide.md:134]: Do we need to include the sentence about allowlist?

File: website/content/en/docs/upgrading/upgrade-guide.md [hunk 7]
```diff
@@ -101,6 +101,7 @@ Below is the full changelog for v1, copied from the [v1 Migration Upgrade Proced
 {{% /alert %}}
 
 * Karpenter changed the name of the `karpenter_cloudprovider_instance_type_price_estimate` metric to `karpenter_cloudprovider_instance_type_offering_price_estimate` to align with the new `karpenter_cloudprovider_instance_type_offering_available` metric. The `region` label was also dropped from the metric, since this can be inferred from the environment that Karpenter is running in.
+* Starting with `0.36.5` Karpenter has enabled conversion webhooks by default to improve the v1 migration experience.
```
Comment (engedaam) [inline website/content/en/docs/upgrading/upgrade-guide.md:104]: We should include these notices in the other version of the docs

File: website/content/en/v0.36/upgrading/upgrade-guide.md (hunk 4 above)
Comment (rschalo) [reply website/content/en/v0.36/upgrading/upgrade-guide.md:28]: It was one AI we had but I think now that we are enabling the webhooks by default it isn't as risky to leave in the guide. I'll put it back.

File: website/content/en/docs/upgrading/upgrade-guide.md (hunk 3 above)
Comment (rschalo) [reply website/content/en/docs/upgrading/upgrade-guide.md:91]: Typo - fixing.

File: website/content/en/docs/upgrading/upgrade-guide.md [hunk 8]
```diff
@@ -101,6 +102,7 @@ Below is the full changelog for v1, copied from the [v1 Migration Upgrade Proced
 {{% /alert %}}
 
 * Karpenter changed the name of the `karpenter_cloudprovider_instance_type_price_estimate` metric to `karpenter_cloudprovider_instance_type_offering_price_estimate` to align with the new `karpenter_cloudprovider_instance_type_offering_available` metric. The `region` label was also dropped from the metric, since this can be inferred from the environment that Karpenter is running in.
+* Starting with `0.36.5` Karpenter has enabled conversion webhooks by default to improve the v1 migration experience. If working with a cluster with a network policy that blocks Ingress, ports 8001, 8443 will need to be allowlisted.
```
Comment (jonathan-innis) [inline website/content/en/docs/upgrading/upgrade-guide.md:105]: Realistically, folks need to allow-list all of the ports exposed on the service -- so we should probably also include ports 8000 and 8081 here as well

File: website/content/en/docs/upgrading/upgrade-guide.md [hunk 9]
```diff
@@ -109,6 +111,7 @@ Below is the full changelog for v1, copied from the [v1 Migration Upgrade Proced
 {{% /alert %}}
 
 * Karpenter OCI tags and Helm chart version are now valid semantic versions, meaning that the `v` prefix from the git tag has been removed and they now follow the `x.y.z` pattern.
+* Starting with `0.35.8` Karpenter has enabled conversion webhooks by default to improve the v1 migration experience. If working with a cluster with a network policy that blocks Ingress, ports 8001, 8443 will need to be allowlisted.
```
Comment (jonathan-innis) [inline website/content/en/docs/upgrading/upgrade-guide.md:114]: Same comment here -- we should apply the recommendation for ports to each of these statements IMO

**Review by jonathan-innis — APPROVED** [review]

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**